python/cudf/cudf/core/series.py

```diff
@@ -1592,7 +1592,9 @@ def label_encoding(self, cats, dtype=None, na_sentinel=-1):
         codes = Series(cupy.arange(len(cats), dtype=dtype))
 
         value = DataFrame({"value": cats, "code": codes})
-        codes = DataFrame({"value": self.copy(), "order": order})
+        codes = DataFrame(
+            {"value": self.copy()._data.columns[0], "order": order}
```

Comment:
Do you need a deep copy here? Can this be shallow?

Reply:
yep, that can be a shallow copy - Changed it to shallow.